packages/components/src/color-palette/test/index.js

```diff
@@ -17,102 +18,146 @@ describe( 'ColorPalette', () => {
 		);
-		activeButton.simulate( 'click' );
 
 		expect( onChange ).toHaveBeenCalledTimes( 1 );
 		expect( onChange ).toHaveBeenCalledWith( undefined );
 	} );
 
-	test( 'should call onClick on an inactive button', () => {
-		const inactiveButton = buttons
-			.findWhere( ( button ) => ! button.hasClass( 'is-pressed' ) )
-			.first();
-		inactiveButton.simulate( 'click' );
+	test( 'should call onClick on an inactive button', async () => {
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
+
+		render(
+			<ColorPalette
+				colors={ colors }
+				value={ currentColor }
+				onChange={ onChange }
+			/>
+		);
+
+		await user.click(
+			screen.getAllByRole( 'button', {
+				name: /^Color:/,
+				pressed: false,
+			} )[ 0 ]
+		);
 
 		expect( onChange ).toHaveBeenCalledTimes( 1 );
 	} );
 
-	test( 'should call onClick with undefined, when the clearButton onClick is triggered', () => {
-		const clearButton = wrapper.find( 'ButtonAction button' );
+	test( 'should call onClick with undefined, when the clearButton onClick is triggered', async () => {
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
 
-		expect( clearButton ).toHaveLength( 1 );
+		render(
+			<ColorPalette
+				colors={ colors }
+				value={ currentColor }
+				onChange={ onChange }
+			/>
+		);
 
-		clearButton.simulate( 'click' );
+		await user.click( screen.getByRole( 'button', { name: 'Clear' } ) );
 
 		expect( onChange ).toHaveBeenCalledTimes( 1 );
 		expect( onChange ).toHaveBeenCalledWith( undefined );
 	} );
 
 	test( 'should allow disabling custom color picker', () => {
-		expect(
-			shallow(
-				<ColorPalette
-					colors={ colors }
-					disableCustomColors={ true }
-					value={ currentColor }
-					onChange={ onChange }
-				/>
-			)
-		).toMatchSnapshot();
-	} );
+		const { container } = render(
+			<ColorPalette
+				colors={ colors }
+				disableCustomColors
+				value={ currentColor }
+				onChange={ onChange }
+			/>
+		);
 
-	describe( 'Dropdown', () => {
-		const dropdown = wrapper.find( 'Dropdown' );
+		expect( container ).toMatchSnapshot();
+	} );
 
-		test( 'should render it correctly', () => {
-			expect( dropdown ).toMatchSnapshot();
+	test( 'should render dropdown and its content', async () => {
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
 		} );
 
-		describe( '.renderToggle', () => {
-			const isOpen = true;
-			const onToggle = jest.fn();
-
-			const renderedToggleButton = mount(
-				dropdown.props().renderToggle( { isOpen, onToggle } )
-			);
-
-			test( 'should render dropdown content', () => {
-				expect( renderedToggleButton ).toMatchSnapshot();
-			} );
+		render(
+			<ColorPalette
+				colors={ colors }
+				value={ currentColor }
+				onChange={ onChange }
+			/>
+		);
 
-			test( 'should call onToggle on click.', () => {
-				renderedToggleButton.find( 'button' ).simulate( 'click' );
+		await user.click(
```

Comment:
Should we add a spy for the `onToggle` callback prop and make sure that `.toHaveBeenCalledTimes( 1 )`, like in the original test?

Reply:
Hm, how would you recommend this to happen without modifications to the original component?

Reply:
My bad, I didn't realize that the `Dropdown` component is an implementation detail, internal to `ColorPalette` and therefore it's not possible to pass a spy for `onToggle` 😅